src/services/users.js

```diff
@@ -0,0 +1,178 @@
+const { User } = require('../models');
+const { UserRoles } = require('../models');
+const { UserStatuses } = require('../models');
+const bcrypt = require('bcryptjs');
+
+class UsersService {
+  async createUser (req, res) {
+    await this.#createBaseUser(req, res, 'User');
+  }
+
+  async createAdmin (req, res) {
+    await this.#createBaseUser(req, res, 'Admin');
+  }
+
+  async getUsers (req, res) {
+    const { _id, name, email, status, role, skip, take } = req.query;
+
+    const query = {};
+    if (_id) query._id = _id;
+    if (name) query.name = name;
+    if (email) query.email = email;
+    if (status) query.status = status;
+    if (role) query.role = role;
+    if (skip) query.skip = skip;
+    if (take) query.take = take;
+
+    try {
+      const users = await User.find(query)
+        .select('-password')
+        .skip(+query.skip || 0)
+        .limit(+query.take || 50)
+        .exec();
+
+      const count = await User.find(query).countDocuments().exec();
+
+      return {
+        total_size: count,
+        list: users ? res.json(users) : res.json([])
+      };
+    } catch (err) {
+      return res.boom.badRequest();
+    }
+  }
+
+  async updateUser (req, res) {
+    try {
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      const currentStatus = user.status;
```

Comment:
Throw not found error here if user doesn't exist

Reply:
fixed